src/tests/JIT/Regression/JitBlue/Runtime_78891/Runtime_78891.cs

```diff
@@ -0,0 +1,75 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+using Xunit;
+
+// This test is to ensure that an assertion does not occur in the JIT.
+public class Runtime_78891
+{
+    public class C0
+    {
+        public long F1;
+    }
+
+    public struct S5
+    {
+        public bool F1;
+        public int F2;
+        public C0 F4;
+        public short F5;
+        public ulong F6;
+        public uint F7;
+    }
+
+    public class Program
+    {
+        public static S5 s_48;
+        public static int Start()
+        {
+            var vr2 = new S5();
+            var vr3 = new S5();
+            try
+            {
+                M59(vr2, vr3);
+                return 0;
+            }
+            catch (NullReferenceException)
+            {
+                // We expect a null reference exception to occur, return 100.
+                return 100;
+            }
```

Comment:
optional: Test methods do not have to return `int` if failure is going to be a crash or can be asserted in some way. There is some debate about the cruft that some of the assertion methods add, but Assert.Throws is a pretty good one.

Typically, these sorts of tests have NoInlining on the method in question (M59 here?).  It's possible that it is now inlined (though I guess not with the finally... though the body isn't going to throw...), destroying the repro.

Anyways, something like `Assert.Throws<NullReferenceException>(() => M59(vr2, vr3))` ought to be sufficient here.

Reply:
Yes, that would be better - guess it's just an old habit writing `return 100` to make it pass :)